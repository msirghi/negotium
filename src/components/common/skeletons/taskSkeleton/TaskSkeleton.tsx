import { Box } from '@mui/system';
import { Skeleton } from '@mui/material';
import { makeStyles } from '@mui/styles';
import { Row } from '../../utilities/row/Row';

const useStyles = makeStyles({
  root: {
    height: '100vh',
  },
  rowItem: {
    marginLeft: 10,
  },
});

export const TaskSkeleton = () => {
  const classes = useStyles();

  return (
    <Box>
      <Skeleton variant="text" width={100} height={40} />
      {[1, 2, 3, 4, 5].map((item) => {
        return (
          <Row key={item}>
            <Skeleton animation="wave" variant="text" width={28} height={40} />
            <Skeleton
              animation="wave"
              variant="text"
              width={'100%'}
              className={classes.rowItem}
              height={40}
            />
          </Row>
        );
      })}
    </Box>
  );
};
