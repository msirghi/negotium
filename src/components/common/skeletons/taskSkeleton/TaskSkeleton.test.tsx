import renderer from 'react-test-renderer';
import { TaskSkeleton } from './TaskSkeleton';
import { mount } from 'enzyme';
import { Skeleton } from '@mui/material';

describe('TaskSkeleton', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<TaskSkeleton />);
    expect(tree).toMatchSnapshot();
  });

  it('should render skeletons', () => {
    const wrapper = mount(<TaskSkeleton />);
    expect(wrapper.find(Skeleton)).not.toHaveLength(0);
  });
});
