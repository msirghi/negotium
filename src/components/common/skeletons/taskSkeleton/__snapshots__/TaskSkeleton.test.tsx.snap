// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskSkeleton should match the snapshot 1`] = `
<div
  className="MuiBox-root css-0"
>
  <span
    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
    style={
      Object {
        "height": 40,
        "width": 100,
      }
    }
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "initial",
      }
    }
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": 28,
        }
      }
    />
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave makeStyles-rowItem-2 css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": "100%",
        }
      }
    />
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "initial",
      }
    }
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": 28,
        }
      }
    />
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave makeStyles-rowItem-2 css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": "100%",
        }
      }
    />
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "initial",
      }
    }
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": 28,
        }
      }
    />
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave makeStyles-rowItem-2 css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": "100%",
        }
      }
    />
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "initial",
      }
    }
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": 28,
        }
      }
    />
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave makeStyles-rowItem-2 css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": "100%",
        }
      }
    />
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "initial",
      }
    }
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": 28,
        }
      }
    />
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave makeStyles-rowItem-2 css-1dk1h2u-MuiSkeleton-root"
      style={
        Object {
          "height": 40,
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
